### PR TITLE
Parallelize commands in bundle-wheel.sh

### PR DESCRIPTION
**Pull request by JanuszL**
- accelerates wheel binary processing and creation by parallelize commands inside

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It parallelize commands in bundle-wheel.sh

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     accelerates wheel binary processing and creation by parallelize commands inside
 - Affected modules and functionalities:
     bundle-wheel.sh
 - Key points relevant for the review:
     Check if it is not too convoluted?
 - Validation and testing:
     current test applies
 - Documentation (including examples):
     NA

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
It reduces CI build time by ~3 minutes.

File: dali/python/bundle-wheel.sh [hunk 1]
```diff
@@ -78,14 +78,16 @@ fname_with_sha256() {
...
+    RECORD_FILE=$2
     if echo $FPATH | grep RECORD >/dev/null 2>&1; then
     # if the RECORD file, then
-    echo "$FPATH,,"
+    result="$FPATH,,"
     else
     HASH=$(openssl dgst -sha256 -binary $FPATH | openssl base64 | sed -e 's/+/-/g' | sed -e 's/\//_/g' | sed -e 's/=//g')
     FSIZE=$(ls -nl $FPATH | awk '{print $5}')
-    echo "$FPATH,sha256=$HASH,$FSIZE"
+    retult="$FPATH,sha256=$HASH,$FSIZE"
```
Comment (jantonguirao) [inline dali/python/bundle-wheel.sh:88]:
```suggestion
    result="$FPATH,sha256=$HASH,$FSIZE"
```

File: dali/python/bundle-wheel.sh [hunk 2]
```diff
@@ -78,14 +78,16 @@ fname_with_sha256() {
...
     fi
+    flock /tmp/dali_rec.lock echo $retult>>$RECORD_FILE
```
Comment (jantonguirao) [inline dali/python/bundle-wheel.sh:90]:
```suggestion
    flock /tmp/dali_rec.lock echo $result>>$RECORD_FILE
```

**Review by jantonguirao — APPROVED** [review]

File: dali/python/bundle-wheel.sh [hunk 3]
```diff
@@ -130,26 +134,21 @@ for filepath in "${DEPS_LIST[@]}"; do
...
-
     echo "Patching DT_SONAME field in $patchedpath"
     patchelf --set-soname $patchedname $TMPDIR/$patchedpath
+}
+
+echo "Patching DT_SONAMEs..."
+for filepath in "${DEPS_LIST[@]}"; do
+    copy_and_patch $filepath
 done
+wait
```
Comment (mzient) [inline dali/python/bundle-wheel.sh:145]: What is it waiting for?

File: dali/python/bundle-wheel.sh [hunk 4]
```diff
@@ -130,26 +134,21 @@ for filepath in "${DEPS_LIST[@]}"; do
...
-        continue
-    fi
-
     echo "Patching DT_SONAME field in $patchedpath"
     patchelf --set-soname $patchedname $TMPDIR/$patchedpath
+}
+
+echo "Patching DT_SONAMEs..."
+for filepath in "${DEPS_LIST[@]}"; do
+    copy_and_patch $filepath
```
Comment (mzient) [inline dali/python/bundle-wheel.sh:143]: Shouldn't this (or the patchelf inside) be run in background?

**Review by mzient — APPROVED** [review]
Fix the typo and (optionally) run 1st patchelf in background. Otherwise OK.

**Review by awolant — APPROVED** [review]

File: dali/python/bundle-wheel.sh (hunk 4 above)
Comment (JanuszL) [reply dali/python/bundle-wheel.sh:144]: I cannot run the whole function as it populates `original` and `patched` tables. When it runs in bg it creates separate processes it won't affect the calling process scope.

File: dali/python/bundle-wheel.sh (hunk 2 above)
Comment (JanuszL) [reply dali/python/bundle-wheel.sh]: Done

File: dali/python/bundle-wheel.sh (hunk 3 above)
Comment (JanuszL) [reply dali/python/bundle-wheel.sh:146]: Fixed by putting patchelf in bg.

**Comment by JanuszL** [comment]
!build

**Pull request merged**